GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Services;
+using Google.Apis.Storage.v1.Data;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Apis.Storage.v1.ClientWrapper
+{
+    /// <summary>
+    /// Wrapper around <see cref="StorageService"/> to provide simpler operations.
+    /// </summary>
+    public class StorageClient
+    {
+        /// <summary>
+        /// The underlying storage service object used by this client.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="StorageClient"/> class only provides convenience operations built on top of
+        /// an existing service object. Any more complex operations which are not supported by this wrapper may wish
+        /// to use the same service object as the wrapper, in order to take advantage of its configuration (for authentication,
+        /// application naming etc).
+        /// </remarks>
+        public StorageService Service { get; }
+
+        /// <summary>
+        /// Constructs a new client wrapping the given <see cref="StorageService"/>.
+        /// </summary>
+        /// <param name="service">The service to wrap. Must not be null.</param>
+        public StorageClient(StorageService service)
+        {
+            Service = service;
+        }
+
+        /// <summary>
+        /// Constructs a new client by creating a <see cref="StorageService"/> for the given <see cref="BaseClientService.Initializer"/>.
+        /// </summary>
+        /// <param name="initializer">The initializer to use in the service. Must not be null.</param>
+        public StorageClient(BaseClientService.Initializer initializer) : this(new StorageService(initializer))
```

Comment:
But it's not from application credentials - it's for any initializer. The point about FromDefaultApplicationCredentials being a factory method is:

- Factory methods can be asynchronous; constructors can't. We don't need *this* to be asynchronous, unlike fetching the default application credentials.
- There's no parameter with an appropriate type to make it obvious what the client would be constructed with; here, anyone who has a BaseClientService.Initializer probably knows what it's for anyway, so there'd be no benefit in spelling it out further.

I can add more documentation, of course...